### ISSUE TITLE
Download Latest/Edited SimBrief OFP Pack with API

### ISSUE THREAD

**Issue by FatihKoz** [nabeelio/phpvms#1226]
**Is your feature request related to a problem? Please describe.**
No, it is a new feature request.

**Describe the feature you'd like**
As you may know, it is possible to edit a SimBrief OFP once it is generated and API (according to its docs) allows re-downloading an edited ofp pack.

Is it possible to add this feature to our own API integration so we would be able to edit a generated ofp (via SimBrief interface of course) and then with a click of a button we can download that latest ofp to our database (with updating current record)

We can build a `static_id` as per SimBrief requirements easily for generated flights with our current form like 
`$user->id."_".$flight->id` or (if it needs a shorter string) `$user->id."_".$flight->number` 
then we can use that static_id we created to pull the latest/edited/regenerated ofp.  

**Describe alternatives you've considered**
None, continue using current code, edit if needed and download manually and use that downloaded file with vmsacars.

**Comment by FatihKoz** [comment]
If you find this useful or applicable to core, just let me know ... I already have it working , able to send a full pr.

**Comment by nabeelio** [comment]
Yeah, let's get this into core, thanks.

**Comment by FatihKoz** [comment]
My pleasure, will do a test with the default theme :) Config, controller
and service changes are working fine but I need to be sure it works on pure
default theme too 'cause I used a modal to open SimBrief webpage to edit
and keep our button to download in the same window.

Expect a new pull request soon

**Comment by nabeelio** [comment]
Is that static ID different from the one that's generated? Because we pass an ID to it on the original generation and that forms the SB ID. But if it's different, use something like `{user-id}-{flight-id}-{hash-id}` - where the hash ID comes from:

```php
use App\Support\Utils;

Utils::generateNewId(10);
```

See what the max length is from SB, and maybe adjust the above-generated ID to fill in the remainder of that length.

**Comment by FatihKoz** [comment]
Yes it is something different, needs to be sent when we are sending the
form fields with the api I think. Maybe Derek wanted to have something more
basic or thought something else for it, when I checked our current requests
and returned xml I saw that the static_id field was empty, thus added a new
hidden field to the form
`$user->ident."_".flight->airline->icao.$flight->flight_number`

I do not think that it needs to be that unique (with hash) 'cause it is
always loading the last ofp that user generated, even the static id seems
pointless but we may never know the inner details of course.Maybe Derek
needs to have some separation while selecting ofp's generated via api and
the ones generated from web interface for the same user. Like he generates
one via api, then generates another (totally different one) via web
interface, in this scenario that static id may be needed to identify the
api generated one.

Anyway, pr is ready, sending soon but I have little problem.

When I use config to read the url and then use str_replace to change it my
test server generates an error and stops.
`[2021-06-06 13:18:48] prod.ERROR: simplexml_load_string(): Entity: line 1:
parser error : Start tag expected, '&lt;' not found
{"userId":3,"exception":"[object] (ErrorException(code: 0):
simplexml_load_string(): Entity: line 1: parser error : Start tag expected,
'&lt;' not found at /home/u9681834/
vms7.turksim.org/app/Services/SimBriefService.php:73)
[stacktrace]`

I am using the same code at my live server, which works fine.
I just dashed out that part and defined the $uri in the service file
directly for you to examine.
